cmd/create_data_stream_test.go

```diff
@@ -0,0 +1,97 @@
+	assert.Equal(t, "title", questions[1].Name)
+	assert.IsType(t, &survey.Input{}, questions[1].Prompt)
+	assert.Equal(t, "type", questions[2].Name)
+	assert.IsType(t, &survey.Select{}, questions[2].Prompt)
+}
+
+func TestGetSurveyQuestionsForVersion_EqualSemver3_2_0(t *testing.T) {
+	version := semver.MustParse("3.2.0")
+	questions := getInitialSurveyQuestionsForVersion(version)
+
+	require.Len(t, questions, 4, "should return 4 questions for spec version >= 3.2.0")
+
+	assert.Equal(t, "subobjects", questions[3].Name)
+	assert.IsType(t, &survey.Confirm{}, questions[3].Prompt)
+}
+
+func TestGetSurveyQuestionsForVersion_AboveSemver3_2_0(t *testing.T) {
+	version := semver.MustParse("3.3.0")
+	questions := getInitialSurveyQuestionsForVersion(version)
+
+	require.Len(t, questions, 4, "should return 4 questions for spec version > 3.2.0")
+
+	assert.Equal(t, "subobjects", questions[3].Name)
+	assert.IsType(t, &survey.Confirm{}, questions[3].Prompt)
+}
+
+func TestCreateDataStreamDescriptorFromAnswers_SubobjectsFalseForSpecVersionBelow3_2_0(t *testing.T) {
+	specVersion := semver.MustParse("3.1.0")
+	answers := newDataStreamAnswers{
+		Name:       "test_stream",
+		Title:      "Test Stream",
+		Type:       "logs",
+		Subobjects: false,
+	}
+	descriptor := createDataStreamDescriptorFromAnswers(answers, "/tmp/package", specVersion)
+
+	assert.Equal(t, "test_stream", descriptor.Manifest.Name)
+	assert.Equal(t, "Test Stream", descriptor.Manifest.Title)
+	assert.Equal(t, "logs", descriptor.Manifest.Type)
+	assert.Equal(t, "/tmp/package", descriptor.PackageRoot)
+	assert.Nil(t, descriptor.Manifest.Elasticsearch)
+}
+
+func TestCreateDataStreamDescriptorFromAnswers_SubobjectsFalseForSpecVersionGTE3_2_0(t *testing.T) {
+	specVersion := semver.MustParse("3.2.0")
+	answers := newDataStreamAnswers{
+		Name:       "test_stream",
+		Title:      "Test Stream",
+		Type:       "logs",
+		Subobjects: false,
+	}
+	descriptor := createDataStreamDescriptorFromAnswers(answers, "/tmp/package", specVersion)
+
+	require.NotNil(t, descriptor.Manifest.Elasticsearch)
+	require.NotNil(t, descriptor.Manifest.Elasticsearch.IndexTemplate)
+	require.NotNil(t, descriptor.Manifest.Elasticsearch.IndexTemplate.Mappings)
+	assert.False(t, descriptor.Manifest.Elasticsearch.IndexTemplate.Mappings.Subobjects)
+}
+
+func TestCreateDataStreamDescriptorFromAnswers_SubobjectsTrueForSpecVersionGTE3_2_0(t *testing.T) {
+	specVersion := semver.MustParse("3.2.0")
+	answers := newDataStreamAnswers{
+		Name:       "test_stream",
+		Title:      "Test Stream",
+		Type:       "logs",
+		Subobjects: true,
+	}
+	descriptor := createDataStreamDescriptorFromAnswers(answers, "/tmp/package", specVersion)
+
+	require.NotNil(t, descriptor.Manifest.Elasticsearch)
+	require.NotNil(t, descriptor.Manifest.Elasticsearch.IndexTemplate)
+	require.NotNil(t, descriptor.Manifest.Elasticsearch.IndexTemplate.Mappings)
+	assert.False(t, descriptor.Manifest.Elasticsearch.IndexTemplate.Mappings.Subobjects)
```

Comment:
I think in my previous suggestion, this assert was incorrect, sorry

```suggestion
	assert.True(t, descriptor.Manifest.Elasticsearch.IndexTemplate.Mappings.Subobjects)
```

Reply:
Or taking into account this condition:
```golang
	if !specVersion.LessThan(semver3_2_0) && !answers.Subobjects {
		manifest.Elasticsearch = &packages.Elasticsearch{
			IndexTemplate: &packages.ManifestIndexTemplate{
				Mappings: &packages.ManifestMappings{
					Subobjects: false,
				},
			},
		}
	}
```
And as subobjects is `true` in this test case, then it should be just nil `descriptor.Manifest.Elasticsearch`? :
```golang
	descriptor := createDataStreamDescriptorFromAnswers(answers, "/tmp/package", specVersion)

	require.Nil(t, descriptor.Manifest.Elasticsearch)
}
```

WDYT?